torch/csrc/autograd/python_engine.cpp

```diff
@@ -237,7 +238,13 @@ PyObject *THPEngine_run_backward(PyObject *self, PyObject *args, PyObject *kwarg
       THPUtils_assert(input_var->cdata.requires_grad(),
           "One of the differentiated Tensors does not require grad");
       if (!grad_fn) {
-        output_edges.emplace_back();
+        // Since input has no grad_accumulator, its guaranteed to be not reachable.
+        // We create a dummy edge with no next edges and a non-nullptr grad_fn so
+        // nodes (e.g., mul when an operand is scalar) that have invalid next_edges
```

Comment:
I'm not sure if I would say invalid here. Maybe empty?
Because even in c++ if you add two Tensors, one that requires grad and one that does not, one of the edge will point nowhere for the Tensor that does not require gradients.

Reply:
Yeah, invalid isn't very clear here

Reply:
Changing it to say: "edge pointing to nullptr"